src/shims/files.rs

```diff
@@ -54,8 +161,9 @@ pub trait FileDescription: std::fmt::Debug + Any {
         throw_unsup_format!("cannot seek on {}", self.name());
     }
 
-    fn close<'tcx>(
-        self: Box<Self>,
+    /// Close the file descriptor. Will only be called when `strong_count` is 1.
+    fn close_unique<'tcx>(
+        self: FileDescriptionRef<Self>,
```

Comment:
This is the only ugly thing I wasn't able to fix... we can't convert a `Rc<dyn ...>` to a `Box<dyn ...>`, nor can we move out the inner value, so there's no better argument type we can use here.

Reply:
I found a way to also avoid this. :)  So overall the PR doesn't add any new `unwrap`, and removes a bunch of previously existing ones. :D